sqe-api-server/Helpers/InputValidation.cs

```diff
@@ -74,14 +106,63 @@ private static string _validatePolygon(string wktPolygon, string entityName, boo
             if (string.IsNullOrEmpty(wktPolygon))
                 throw new StandardExceptions.InputDataRuleViolationException("The submitted WKT POLYGON is empty");
 
-
+            var polygon = default(Geometry);
             // Try loading the polygon
             try
             {
-                var polygon = wkr.Read(wktPolygon);
+                polygon = wkr.Read(wktPolygon);
                 // If it is valid, return it
                 if (polygon.IsValid)
                     return polygon.Normalized().ToString(); // Always normalize the polygon
+
+                // It is invalid, but could be repaired as a binary representation
+                // Throw an error if no request to fix it has been made
+                if (!fix)
+                    throw new StandardExceptions.InputDataRuleViolationException("The submitted WKT POLYGON is invalid, try using the API's repair-wkt-polygon endpoint to fix it");
+
+                // Try repairing the binary version of the polygon
+                var wkb_in = polygon.AsBinary(); // Get the binary data
+                // Instantiate the variable to process the response from the FFI
+                var bin = new BinaryData();
+                var wkbData = new Span<byte>();
+
+                // Reading and writing pointers is unsafe
+                try
+                {
+                    unsafe
```

Comment:
I am checking through this again now.  I agree with you about the unsafe code.  It may have had something to do with freeing the memory, but I am going back now and testing out the approaches without unsafe.  Then we will either get rid of unsafe (I hope to do that) or we will have a nice detailed note as to why we had to use unsafe.  Thanks.